controller/resource.go

```diff
@@ -37,12 +43,57 @@ func (c *ResourceController) Read(ctx *app.ReadResourceContext) error {
 
 // Register runs the register action.
 func (c *ResourceController) Register(ctx *app.RegisterResourceContext) error {
-	// ResourceController_Register: start_implement
 
-	// Put your logic here
+	// TODO validate the PAT here
 
-	// ResourceController_Register: end_implement
-	return nil
+	var res *resource.Resource
+
+	err := application.Transactional(c.db, func(appl application.Application) error {
+
+		// Lookup or create the resource type
+		resourceType, err := appl.ResourceTypeRepository().LookupOrCreate(ctx, ctx.Payload.Name)
```

Comment:
Will we allow to create resource type by resource servers? Or should we pre-define them?

Reply:
I think it's easiest to just allow the resource servers to control this, no need for us to pre-define the resource types.  Also this makes it compliant with the UMA spec.